SteamKit2/SteamKit2/Util/HardwareUtils.cs

```diff
@@ -309,22 +351,22 @@ public void Set3B3( string value )
                 this.KeyValues["3B3"].Value = value;
             }
 
+            
+            // 333 is some sort of user supplied data and is currently unused
             public void Set333( string value )
             {
                 this.KeyValues["333"] = new KeyValue( value: value );
             }
         }
-
-
+        
         static Task<MachineID>? generateTask;
-
-
-        public static void Init()
+        
+        internal static void Init(IMachineInfoProvider provider)
```

Comment:
If a user sets a provider, we should kick off the generator task then, not wait for a later point.